drivers/entropy/entropy_gecko_trng.c

```diff
@@ -51,6 +51,35 @@ static int entropy_gecko_trng_get_entropy(struct device *dev, u8_t *buffer,
 	return 0;
 }
 
+static int entropy_gecko_trng_get_entropy_isr(struct device *dev, u8_t *buf,
+					u16_t len, u32_t flags)
+{
+
+	if ((flags & ENTROPY_BUSYWAIT) == 0U) {
+
+		/* No busy wait; return whatever data is available. */
+		size_t count;
+		size_t available = TRNG0->FIFOLEVEL * 4;
+
+		if (available == 0) {
+			return -EINVAL;
```

Comment:
`-ENODATA` probably is the best error return. Though error return is not consistent across drivers :/

Reply:
done